crates/miden-tx/src/executor/exec_host.rs

```diff
@@ -467,72 +487,197 @@ where
         &mut self,
         process: &ProcessState,
     ) -> impl FutureMaybeSend<Result<Vec<AdviceMutation>, EventError>> {
-        let event_id = EventId::from_felt(process.get_stack_item(0));
-
-        // TODO: Eventually, refactor this to let TransactionEvent contain the data directly, which
-        // should be cleaner.
-        let event_handling_result = self.base_host.handle_event(process, event_id);
+        let stdlib_event_result = self.base_host.handle_stdlib_events(process);
+        let tx_event_result = TransactionEvent::extract(&self.base_host, process);
```

Comment:
Right now, this would execute `TransactionEvent::extract()` regardless of whether the event was handled by the standard library handlers. I'd probably make this conditional - something like:

```Rust
let stdlib_event_result = self.base_host.handle_stdlib_events(process);
let tx_event_result = match stdlib_event_result {
    Ok(None) => TransactionEvent::extract(&self.base_host, process),
    _ => None,
};
```